pandas/tests/scalar/timestamp/test_timestamp.py

```diff
@@ -223,6 +225,31 @@ def test_resolution(self):
         assert dt.as_unit("ms").resolution == Timedelta(milliseconds=1)
         assert dt.as_unit("s").resolution == Timedelta(seconds=1)
 
+    @pytest.mark.parametrize(
+        "date_string, expected_true, expected_false",
+        [
+            ("0000-2-29", 1, 2),
+            ("0000-3-1", 2, 3),
+            ("1582-10-14", 3, 4),
+            ("1582-10-15", 4, 5),
+            ("1752-01-01", 5, 6),
+            ("2023-06-18", 6, 0),
+        ],
+    )
+    def test_dow_historic(self, date_string, expected_true, expected_false):
+        # GH 53738
+        dt = Timestamp(date_string)
+        dow = dt.weekday()
+        assert dow == expected_true
+        assert not dow == expected_false
+
+    @given(DATETIME_IN_PD_TIMESTAMP_RANGE_NO_TZ)
```

Comment:
I think we should just use `st.datetimes` (where `st` is hypothesis.strategies), and then:
- convert that to a pandas Timestamp
- compare Timestamp.weekday with the standard library's .weekday()

`DATETIME_IN_PD_TIMESTAMP_RANGE_NO_TZ` only goes from about 1700 to about 2300 - I think there's a separate issue about that. But anyway, until that's fixed, let's use the full range of what the standard library offers

Reply:
Done!